NPE when resolving binding

The issue happens when editing a Java file that isn't on the project's classpath.

See https://github.com/Microsoft/vscode-java-debug/issues/150

[test.zip](https://github.com/eclipse/eclipse.jdt.ls/files/1492380/test.zip)

```
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:604)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:443)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157)
Caused by: java.lang.NullPointerException
	at org.eclipse.jdt.ls.core.internal.text.correction.QuickAssistProcessor.getUniqueMethodName(QuickAssistProcessor.java:526)
	at org.eclipse.jdt.ls.core.internal.text.correction.QuickAssistProcessor.getExtractMethodProposal(QuickAssistProcessor.java:251)
	at org.eclipse.jdt.ls.core.internal.text.correction.QuickAssistProcessor.getAssists(QuickAssistProcessor.java:140)
	at org.eclipse.jdt.ls.core.internal.handlers.CodeActionHandler.getCodeActionCommands(CodeActionHandler.java:80)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$11(JDTLanguageServer.java:401)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
	... 6 more
```

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/text/correction/QuickAssistProcessor.java
@@ -523,6 +523,9 @@ public class QuickAssistProcessor {
 				return suggestedName;
 			}
 			IType type = (IType) typeBinding.getJavaElement();
+			if (type == null) {
+				return suggestedName;
+			}
 			IMethod[] methods = type.getMethods();
 
 			int suggestedPostfix = 2;

--- a/org.eclipse.jdt.ls.tests/projects/eclipse/hello/nopackage/Test.java
+++ b/org.eclipse.jdt.ls.tests/projects/eclipse/hello/nopackage/Test.java
@@ -1,0 +1,5 @@
+public class Test{
+    public static void main(String[] args) {
+        System.out.println("test");
+    }
+}

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/CodeActionHandlerTest.java
@@ -10,9 +10,12 @@
  *******************************************************************************/
 package org.eclipse.jdt.ls.core.internal.handlers;
 
+import java.net.URI;
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
+import org.eclipse.core.runtime.NullProgressMonitor;
 import org.eclipse.jdt.core.ICompilationUnit;
 import org.eclipse.jdt.core.JavaModelException;
 import org.eclipse.jdt.core.compiler.IProblem;
@@ -25,6 +28,7 @@ import org.eclipse.lsp4j.CodeActionParams;
 import org.eclipse.lsp4j.Command;
 import org.eclipse.lsp4j.Diagnostic;
 import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4j.Position;
 import org.eclipse.lsp4j.Range;
 import org.eclipse.lsp4j.TextDocumentIdentifier;
 import org.eclipse.lsp4j.WorkspaceEdit;
@@ -97,6 +101,29 @@ public class CodeActionHandlerTest extends AbstractCompilationUnitBasedTest {
 		Assert.assertEquals(1, commands.size());
 		Command c = commands.get(0);
 		Assert.assertEquals(CodeActionHandler.COMMAND_ID_APPLY_EDIT, c.getCommand());
+	}
+
+	@Test
+	public void testCodeAction_exception() throws JavaModelException {
+		URI uri = project.getFile("nopackage/Test.java").getRawLocationURI();
+		ICompilationUnit cu = JDTUtils.resolveCompilationUnit(uri);
+		try {
+			cu.becomeWorkingCopy(new NullProgressMonitor());
+			CodeActionParams params = new CodeActionParams();
+			params.setTextDocument(new TextDocumentIdentifier(uri.toString()));
+			final Range range = new Range();
+			range.setStart(new Position(0, 17));
+			range.setEnd(new Position(0, 17));
+			params.setRange(range);
+			CodeActionContext context = new CodeActionContext();
+			context.setDiagnostics(Collections.emptyList());
+			params.setContext(context);
+			List<? extends Command> commands = server.codeAction(params).join();
+			Assert.assertNotNull(commands);
+			Assert.assertEquals(0, commands.size());
+		} finally {
+			cu.discardWorkingCopy();
+		}
 	}
 
 	@Test

